### PR TITLE
[3.4] Fix nullptr dereference in SynchronizeShard.

### PR DESCRIPTION
**Pull request by danielhlarkin**
Should address https://github.com/arangodb/release-3.4/issues/142.

### CODE REVIEW

**Comment by danielhlarkin** [comment]
http://jenkins01.arangodb.biz:8080/job/arangodb-matrix-pr/1972/

File: arangod/Cluster/SynchronizeShard.cpp [hunk 1]
```diff
@@ -157,7 +157,7 @@ static arangodb::Result getReadLockId (
 
   auto result = comres->result;
 
-  if (result != nullptr && result->getHttpReturnCode() == 200) {
+  if (result != nullptr || result->getHttpReturnCode() == 200) {
```
Comment (jsteemann) [inline arangod/Cluster/SynchronizeShard.cpp]: this is wrong now.

File: arangod/Cluster/SynchronizeShard.cpp [hunk 2]
```diff
@@ -356,7 +356,7 @@ static arangodb::Result cancelBarrier(
 
   if (comres->status == CL_COMM_SENT) {
     auto result = comres->result;
-    if (result != nullptr && result->getHttpReturnCode() != 200 &&
+    if (result != nullptr || result->getHttpReturnCode() != 200 &&
```
Comment (jsteemann) [inline arangod/Cluster/SynchronizeShard.cpp]: this is wrong now.

File: arangod/Cluster/SynchronizeShard.cpp [hunk 3]
```diff
@@ -415,7 +415,7 @@ arangodb::Result SynchronizeShard::getReadLock(
       std::unordered_map<std::string, std::string>(), timeout);
 
     auto result = putres->result;
-    if (result != nullptr && result->getHttpReturnCode() == 200) {
+    if (result != nullptr || result->getHttpReturnCode() == 200) {
```
Comment (jsteemann) [inline arangod/Cluster/SynchronizeShard.cpp]: this is wrong now.

**Pull request merged**